src/eids.js

```diff
@@ -56,15 +56,15 @@ export class EidRewriter {
     this.idExemptButPassToChildren = [
       'arguments', 'background', 'conclusions', 'decision', 'header', 'intro', 'introduction', 'motivation', 'preamble', 'preface', 'remedies', 'wrapUp',
     ];
-    this.counters = {};
-    this.eIdCounter = {};
+    this.counters = counters || {};
+    this.eIdCounter = eIdCounter || {};
     this.mappings = {};
   }
 
   /** Rewrites the eIds for all nodes in the tree.
    */
   rewriteAllEids (element, prefix = '') {
-    this.reset();
+    this.reset(this.counters, this.eIdCounter);
```

Comment:
this won't reset any more, because it's just doing `this.counters = this.counters` etc.

I think it's easier not to make this change at all and just change the state directly from where we call this class.